vm/actor/tests/reward_actor_test.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod common;
+
+use actor::{
+    reward::{AwardBlockRewardParams, Method},
+    REWARD_ACTOR_ADDR, REWARD_ACTOR_CODE_ID, SYSTEM_ACTOR_ADDR, SYSTEM_ACTOR_CODE_ID,
+};
+use address::Address;
+use common::*;
+use db::MemoryDB;
+use ipld_blockstore::BlockStore;
+use message::UnsignedMessage;
+use vm::{ExitCode, Serialized, TokenAmount, METHOD_CONSTRUCTOR};
+
+fn construct_runtime<BS: BlockStore>(bs: &BS) -> MockRuntime<'_, BS> {
+    let message = UnsignedMessage::builder()
+        .to(REWARD_ACTOR_ADDR.clone())
+        .from(SYSTEM_ACTOR_ADDR.clone())
+        .build()
+        .unwrap();
+    let mut rt = MockRuntime::new(bs, message);
+    rt.caller_type = SYSTEM_ACTOR_CODE_ID.clone();
+    return rt;
+}
+
+#[test]
+fn test_award_block_reward() {
+    let bs = MemoryDB::default();
+    let mut rt = construct_runtime(&bs);
+    construct_and_verify(&mut rt);
+
+    let miner = Address::new_id(1000);
+    let gas_reward = TokenAmount::from(10u8);
+
+    rt.expect_validate_caller_addr(&[SYSTEM_ACTOR_ADDR.clone()]);
+
+    let params = AwardBlockRewardParams {
+        miner: miner,
+        penalty: TokenAmount::from(0u8),
+        gas_reward: gas_reward,
+        ticket_count: 0,
+    };
+
+    let prev_state = rt.state.clone();
+
+    let call_result = rt.call(
+        &*REWARD_ACTOR_CODE_ID,
+        Method::AwardBlockReward as u64,
+        &Serialized::serialize(&params).unwrap(),
+    );
+
+    // Expecting insufficdent funds, revert state
+    assert_eq!(
+        ExitCode::ErrInsufficientFunds,
+        call_result.unwrap_err().exit_code()
+    );
+    rt.state = prev_state;
```

Comment:
Is there a reason for saving the mock rt state and then resetting it at the end? Tests still pass without those two lines and I can't understand why they'd be there

Reply:
I think you don't need it for this case but there are tests where not reverting the state would fail. I remember writing a test for the init actor where not reverting the state cause it to resolve a address it shouldn't have been able to. In any case I'm just following what they got in the go tests